feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/staking/start/RealCompoundStartStakingInteractor.kt

```diff
@@ -0,0 +1,75 @@
+package io.novafoundation.nova.feature_staking_impl.domain.staking.start
+
+import io.novafoundation.nova.common.utils.combine as combineList
+import io.novafoundation.nova.feature_account_api.domain.interfaces.AccountRepository
+import io.novafoundation.nova.feature_staking_impl.domain.era.StakingEraInteractor
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.model.PayoutType
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.model.StartStakingEraInfo
+import io.novafoundation.nova.feature_wallet_api.domain.interfaces.WalletRepository
+import io.novafoundation.nova.feature_wallet_api.domain.model.Asset
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import java.math.BigDecimal
+import java.math.BigInteger
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flatMapLatest
+
+class StartStakingCompoundData(
+    val chain: Chain,
+    val asset: Asset,
+    val availableBalance: BigInteger,
+    val maxEarningRate: BigDecimal,
+    val minStake: BigInteger,
+    val eraInfo: StartStakingEraInfo,
+    val participateInGovernance: Boolean,
+    val participateInGovernanceMinAmount: BigInteger?,
```

Comment:
Better to create sealed class to prohibit situation when `participateInGovernance=false` but `participateInGovernanceMinAmount=Some`